internal/utility/regex.go

```diff
@@ -0,0 +1,17 @@
+package utility
```

Comment:
Per https://go.dev/blog/package-names, let's avoid generic package names like these.

Let's just call this cachedregexp or something similar. We can make the function name match the one in the canonical `regexp` as well.

Reply:
+1. Would also allow you to simplify the other names in the file

possibilities:
`compiledRegexCache` -> `cache`
`CachedRegexMustCompile` -> `MustCompile`